### DISCUSSION TITLE
Is there an option to provide a new input?

### DISCUSSION THREAD

**Discussion by thjaeckle** [quicktype/quicktype#1886] [Q&A]
I found this project by searching for code generators based on JSON data.
Looks very interesting - and now I wonder if quicktype can be enhanced to support code-generationg for other (JSON based) inputs?

Concretely the W3C "Web of Things" standard defines a JSON-LD document named "Thing Description" (specified here: [https://www.w3.org/TR/wot-thing-description11/](https://www.w3.org/TR/wot-thing-description11/)) which describes the intercation possibilities of IoT devices in a standardized way.
In some ways it is comparable with OpenAPI, but with IoT device focus and not only HTTP is supported, but also IoT protocols as MQTT and CoAP.

Would it be possible to provide my own "Input" based on that specification?

**Comment by adam-rocska** [comment]
I’m wrestling the same problem, or at least domain of problems. Did you find any feasible solution?
